nsxt/resource_nsxt_upgrade_run.go

```
@@ -722,3 +812,74 @@ func resourceNsxtUpgradeRunUpdate(d *schema.ResourceData, m interface{}) error {
 func resourceNsxtUpgradeRunDelete(d *schema.ResourceData, m interface{}) error {
 	return nil
 }
+
+func getHostDefaultUpgradeGroup(m interface{}, hostID string) (string, error) {
+	connector := getPolicyConnector(m)
+	hostClient := nsx.NewTransportNodesClient(connector)
+	host, err := hostClient.Get(hostID)
+	if err != nil {
+		return "", err
+	}
+	converter := bindings.NewTypeConverter()
+	base, errs := converter.ConvertToGolang(host.NodeDeploymentInfo, model.HostNodeBindingType())
+	if errs != nil {
+		return "", errs[0]
+	}
+	node := base.(model.HostNode)
+
+	if node.ComputeCollectionId != nil {
+		return *node.ComputeCollectionId, nil
+	}
+
+	// This host is not a part of a compute cluster:
+	// it should be assigned to the 'Group 1 for ESXI' group (this value is hardcoded in NSX)
+	groupClient := upgrade.NewUpgradeUnitGroupsClient(connector)
+	componentType := "HOST"
+	hostGroups, err := groupClient.List(&componentType, nil, nil, nil, nil, nil, nil, nil)
+	if err != nil {
+		return "", err
+	}
+	if hostGroups.Results != nil {
+		for _, group := range hostGroups.Results {
+			if group.DisplayName != nil && *group.DisplayName == hostUpgradeUnitDefaultGroup {
+				return *group.Id, nil
+			}
+		}
+	}
+
+	return "", errors.NotFound{}
+}
+
+func addHostToGroup(m interface{}, groupID, hostID string, isCreate bool) error {
+	connector := getPolicyConnector(m)
+	client := upgrade.NewUpgradeUnitGroupsClient(connector)
+
+	doUpdate := func() error {
+		group, err := client.Get(groupID, nil)
+		if err != nil {
+			return err
+		}
+
+		hostIDs := getUnitIDsFromUnits(group.UpgradeUnits)
+		if slices.Contains(hostIDs, hostID) {
+			return fmt.Errorf("host %s already exists in group %s", hostID, groupID)
```

should we swallow this error?